Sources/Extraction/Filters/LocalHistogram.c

```diff
@@ -18,20 +18,23 @@ Int16Array3D LocalHistogram_Analyze(const BlockMap *blocks, const UInt8Array2D *
     return histogram;
 }
 
-Int16Array3D LocalHistogram_SmoothAroundCorners(const BlockMap *blocks, const Int16Array3D *input)
+Int16Array3D LocalHistogram_SmoothAroundCorners(const Int16Array3D *input)
 {
-    Int16Array3D output;
+    Int16Array3D output = Int16Array3D_Construct(input->sizeX + 1, input->sizeY + 1, input->sizeZ);
```

Comment:
Also a pattern we're using is to take in a pre-allocated pointer for the output array.

i.e. the function signature would be
`void LocalHistogram_SmoothAroundCorners(const Int16Array3D *input, Int16Array3D *output)`

This way the client knows it's their responsibility to allocate the memory and subsequently free it.